test/forward_ref_test.dart

```diff
@@ -30,13 +30,17 @@ main() {
 
     group('sets name on the rendered component as expected', () {
       test('unless the displayName argument is not passed to forwardRef2', () {
-        final ForwardRefTestComponent = forwardRef2((props, ref) {});
+        final ForwardRefTestComponent = forwardRef2((props, ref) {
+          return null;
+        });
```

Comment:
Another nice thing about switching from a `dynamic` return type to `Object?` is that for function components, you get an analysis warning if you forget to return something.